lib/Bitcoin/Crypto/Block.pm

```diff
@@ -71,29 +455,60 @@ Bitcoin::Crypto::Block - Stripped down block instance
 
 	print $next_block->median_time_past;
 
-
 =head1 DESCRIPTION
 
-This is a block instance required for locktime and sequence checks in
-transactions. It is used in L<Bitcoin::Crypto::Transaction/verify> and
-L<Bitcoin::Crypto::Transaction::UTXO/block>.
+This is a complete Bitcoin block implementation that can parse, construct, and
+serialize Bitcoin blocks. It supports all standard Bitcoin block operations
+including transaction management, merkle root calculation, and block validation.
 
-Bitcoin::Crypto does not contain any real blocks implementation. This class
-provides the bare minimum required for checking locktime and sequence.
+The class also provides the functionality required for locktime and sequence
+checks in transactions, as used in L<Bitcoin::Crypto::Transaction/verify> and
+L<Bitcoin::Crypto::Transaction::UTXO/block>.
```

Comment:
This could mention which attributes are not required when the class is used just for transaction verification.